cantools/database/can/signal.py

```diff
@@ -332,9 +331,9 @@ def raw_to_scaled(
         :return:
             The calculated scaled value
         """
-        if decode_choices:
-            with contextlib.suppress(KeyError, TypeError):
-                return self.choices[raw]  # type: ignore[index]
+        if decode_choices and self.choices and raw in self.choices:
+            assert isinstance(raw, int)
```

Comment:
`isinstance` is slow and we shouldn't use `assert` outside of test code.
I'd suggest `return self.choices[cast(int, raw)]`, because the `raw in self.choices` check guarantees, that `raw` is an integer.

Reply:
I've pushed a new version of the patch. I'm a bit out of my depths of where the remaining performance regressions stem from. (IMO they are acceptable, though.) Can you help out?